packages/storybook/src/stories/Typography.stories.tsx

```diff
@@ -174,3 +174,13 @@ export const Uppercase = TemplateAll.bind({})
 Uppercase.args = {
   uppercase: true,
 }
+
+export const WithBreakpoints = Template.bind({})
+WithBreakpoints.args = {
+  variant: {
+    0: 'p4',
+    1000: 'p3',
+    1500: 'p2',
+  },
+  children: 'Variant by breakpoint test (resize your screen for to change my size)',
+}
```

Comment:
So it will basically set p4 styles up to 1000px. From 1000 to 1500, it will be using p3 styles. And from 1500 it will be using p2 styles.